enhancements/sig-architecture/81-addon-lifecycle/README.md

```diff
@@ -37,26 +38,787 @@ proposal's scope.
+}
+```
+
+`ClusterManagementAddOn` removes `SupportedConfigs` and adds `Configuration` section to list the add-on configurations 
+and the rollout strategy when configurations change.
+
+```golang
+// ClusterManagementAddOnSpec provides information for the add-on.
+type ClusterManagementAddOnSpec struct {
+  // addOnMeta is a reference to the metadata information for the add-on.
+  // +optional
+  AddOnMeta AddOnMeta `json:"addOnMeta,omitempty"`
+
+  // configuration list the add-on configurations and the rollout strategy when the configurations change.
+  // +optional
+  Configuration Configuration `json:"configuration,omitempty"`
+}
+
+// Configuration represents a list of the add-on configurations and their rollout strategy.
+type Configuration struct {
+  // configs is a list of add-on configurations.
+  // The add-on configurations for each cluster can be overridden by the configs of the ManagedClusterAddon spec.
+  // +optional
+  Configs []AddOnConfig `json:"configs,omitempty"`
+
+  // The rollout strategy to apply new configs. 
+  // The rollout strategy only watches the listed configs change. 
+  // If the rollout strategy is not defined, the default strategy UpdateAll is used.
+  // If there are configs change during the rollout process, the rollout will start over. For example, configs list 
+  // configA and configB. The change in configA triggers the rollout. If the configB is also
+  // changed before the rollout is complete, the current rollout stops and the rollout starts over.
+  // +optional
+  RolloutStrategy RolloutStrategy `json:"rolloutStrategy,omitempty"`
+}
+
+// RolloutStrategy represents the rollout strategy of the add-on configuration.
+type RolloutStrategy struct {
+  // Type is the type of the rollout strategy, it supports UpdateAll and RollingUpdateWithPlacement:
+  // - UpdateAll: when configs change, apply the new configs to all the clusters.
+  // - RollingUpdateWithPlacement: when configs change, rolling update new configs on the clusters 
+  //   selected by placements. 
+  //   If any of the configs are overridden by ManagedClusterAddOn on the specific cluster, the new configs 
+  //   won't take effect on that cluster.
+  //   This rollout strategy is only responsible for applying new configs. When the strategy is modified or 
+  //   removed, the applied configs won't be deleted from the cluster.
+  // +kubebuilder:validation:Enum=RollingUpdateWithPlacement;UpdateAll
+  // +kubebuilder:default:=UpdateAll
+  // +optional
+  Type string `json:"type"`
+
+  // Rolling update with placement config params. Present only if the type is RollingUpdateWithPlacement.
+  // +optional
+  RollingUpdateWithPlacement *RollingUpdateWithPlacement `json:"rollingUpdateWithPlacement,omitempty"`
+}
+
+// RollingUpdateWithPlacement represents the placement and behavior to rolling update add-on configurations
+// on the selected clusters.
+type RollingUpdateWithPlacement struct {
+  // name of the placement
```

Comment:
How will this intersect with install strategy?  The PR in https://github.com/open-cluster-management-io/enhancements/pull/77 looked pretty good to me and allowed targeting multiple placements in a way that seemed useful.

Having an overall set of placements for the `ClusterManagementAddOn` and a way to specify a different placement (list of placements?) for a canary would match your description.  Is that the intent of this placement?